### PR TITLE
Add exclusion rule support for urls

### PR DESCRIPTION
**Pull request by theogravity**

### CODE REVIEW

**Comment by defunctzombie** [comment]
Add a test for this please

**Comment by theogravity** [comment]
@defunctzombie  added, it seems that the 0.8 node install is failing on the CI though, seems to be independent of my changes

File: lib/fingerprint.js [hunk 1]
```diff
@@ -51,6 +53,21 @@ module.exports = function(opt, hash_provider) {
             return cb();
         }
 
+        // test if the url should be hashed or not
+        if (opt.exclude) {
+            excluded = opt.exclude.some(function (rule) {
```
Comment (defunctzombie) [inline lib/fingerprint.js]: does this function need to be created inside of every onnode call?

File: lib/fingerprint.js [hunk 2]
```diff
@@ -51,6 +53,21 @@ module.exports = function(opt, hash_provider) {
...
+                if (typeof rule === 'string' && orig.indexOf(rule) !== -1) {
+                    return true;
+                } else if (typeof rule === 'object' && rule.test && rule.test(orig)) {
+                    return true;
+                }
+            });
+        }
+
+        if (excluded) {
```
Comment (defunctzombie) [inline lib/fingerprint.js]: You don't need this if statement again here and can just check excluded above in the previous if

File: lib/fingerprint.js (hunk 1 above)
Comment (defunctzombie) [inline lib/fingerprint.js]: This can be `const` if you remove the if statement below

**Review by defunctzombie — CHANGES_REQUESTED** [review]

**Comment by defunctzombie** [comment]
Thanks for adding the tests. I made some comments on the PR. You can ignore the failure for 0.8 as I will just remove that altogether from travis and update the versions this is tested on.

File: lib/fingerprint.js (hunk 1 above)
Comment (theogravity) [inline lib/fingerprint.js]: it does, it depends on `orig` from line 59.

Unless you'd like me to move the function to a higher level and bind `orig` to the context.

**Review by defunctzombie — APPROVED** [review]

**Pull request merged**